### PR TITLE
Improvement/endpoints

### LINKED ISSUE #145: change endpoints to not return children
make `/networks` and `/devices` not return all their children
maybe change current design to a different endpoint

### PR DESCRIPTION
**Pull request by Shiny380**
Resolves #145 
**ALL** `network` and `device` endpoints now accept `with_children` argument in the query string
i.e.
- `/api/modbus/network?with_children=true` will not return all children
- `/api/modbus/network` will not return just the network data

Depends on #152

### CODE REVIEW

File: src/resources/utils.py [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+def model_marshaller_with_children(data: any, args: dict, base_fields: dict, children_fields: dict):
+    with_children = False
+    if 'with_children' in args:
+        try:
+            with_children = bool(strtobool(args['with_children']))
+        except:
+            abort(400, message='Invalid query string')
+
+    if with_children:
+        print('WITH CHILDREN', children_fields)
```
Comment (RaiBnod) [inline src/resources/utils.py:16]: Use logger instead of print.

File: src/resources/resource_device.py [hunk 2]
```diff
@@ -1,40 +1,34 @@
...
-class DeviceResource(Resource):
-    parser = reqparse.RequestParser()
-    for attr in device_all_attributes:
-        parser.add_argument(attr,
-                            type=device_all_attributes[attr]['type'],
-                            required=device_all_attributes[attr].get('required', False),
-                            help=device_all_attributes[attr].get('help', None),
-                            store_missing=False)
+def device_marshaller(data: any, args: dict):
+    return model_marshaller_with_children(data, args, device_all_fields, device_all_fields_with_children)
```
Comment (RaiBnod) [inline src/resources/resource_device.py:10]: We can use decorator for such cases, but fine for now.

**Review by RaiBnod — APPROVED** [review]
Nothing special changes requested, just suggestions. So you can merge this one after your previous pull request gets merged.

File: src/resources/resource_device.py (hunk 2 above)
Comment (Shiny380) [reply src/resources/resource_device.py:10]: nah I didn't use the decorator as I return 2 different marshallings depending on the query string

**Pull request merged**